### PR TITLE
JAMES-2630 Add CassandraAsyncExecutor.executeRows

### PR DESCRIPTION
**Pull request by blackheaven**
https://github.com/linagora/james-project/pull/2156#discussion_r256231489

https://github.com/linagora/james-project/pull/2156#discussion_r256231761

### CODE REVIEW

File: server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/CassandraMailQueueBrowser.java [hunk 1]
```diff
@@ -131,11 +129,11 @@ private Mail toMail(EnqueuedItem enqueuedItem, MimeMessage mimeMessage) {
         return mail;
     }
 
-    private Mono<List<EnqueuedItemWithSlicingContext>> browseSlice(MailQueueName queueName, Slice slice) {
+    private Flux<EnqueuedItemWithSlicingContext> browseSlice(MailQueueName queueName, Slice slice) {
         return
             allBucketIds()
                 .flatMap(bucketId -> browseBucket(queueName, slice, bucketId))
-                .collectSortedList(Comparator.comparing(enqueuedMail -> enqueuedMail.getEnqueuedItem().getEnqueuedTime()));
+                .sort(Comparator.comparing(enqueuedMail -> enqueuedMail.getEnqueuedItem().getEnqueuedTime()));
```
Comment (mbaechler) [inline server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/CassandraMailQueueBrowser.java]: by doing so, we loose the information that it's ordered that was previously encoded in the return type

File: event-sourcing/event-store-cassandra/src/main/java/org/apache/james/eventsourcing/eventstore/cassandra/EventStoreDao.java [hunk 2]
```diff
@@ -97,10 +97,9 @@ private BoundStatement insertEvent(Event event) {
     }
 
     public History getEventsOfAggregate(AggregateId aggregateId) {
-        return cassandraAsyncExecutor.execute(
+        return cassandraAsyncExecutor.executeRows(
```
Comment (mbaechler) [inline event-sourcing/event-store-cassandra/src/main/java/org/apache/james/eventsourcing/eventstore/cassandra/EventStoreDao.java]: I think that we are wrong in this method: events should be ordered by their id before building the history. Would you mind checking we cover that ?

File: server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/CassandraMailQueueBrowser.java (hunk 1 above)
Comment (blackheaven) [reply server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/CassandraMailQueueBrowser.java]: I do not think it's a big deal, it's a private method used in one place.

File: event-sourcing/event-store-cassandra/src/main/java/org/apache/james/eventsourcing/eventstore/cassandra/EventStoreDao.java (hunk 2 above)
Comment (blackheaven) [reply event-sourcing/event-store-cassandra/src/main/java/org/apache/james/eventsourcing/eventstore/cassandra/EventStoreDao.java]: It's already verified in `EventStoreTest.getEventsOfAggregateShouldReturnAppendedEvents()`

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdDAOTest.java [hunk 3]
```diff
@@ -41,8 +43,12 @@
 import reactor.core.publisher.Flux;
 
 class CassandraMessageIdDAOTest {
+    public static final CassandraModule MODULE = CassandraModule.aggregateModules(
+        CassandraMessageModule.MODULE,
+        CassandraSchemaVersionModule.MODULE);
+
     @RegisterExtension
-    static CassandraClusterExtension cassandraCluster = new CassandraClusterExtension(CassandraMessageModule.MODULE);
+    static CassandraClusterExtension cassandraCluster = new CassandraClusterExtension(MODULE);
```
Comment (rouazana) [inline mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdDAOTest.java]: unrelated to this commit?

Comment (blackheaven) [reply mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdDAOTest.java]: Necessary to run only this test class

Comment (rouazana) [reply mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdDAOTest.java]: ok, but next time you could have separated this into a proper commit

Comment (blackheaven) [reply mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdDAOTest.java]: At one point we either have to either do it everywhere, on add a neat helper.

**Comment by rouazana** [comment]
merged

**Pull request closed without merging**